consensus/ethash: TestRemoteMutliNotify is flaky

Rel https://github.com/ethereum/go-ethereum/pull/17526

```
> go test -run '^TestRemoteMultiNotify$' -count 1 ./consensus/ethash
ok      github.com/ethereum/go-ethereum/consensus/ethash        0.171s
ubp52 ~/go/src/github.com/ethereum/go-ethereum merge/drossix-blue *%
> go test -run '^TestRemoteMultiNotify$' -count 100 ./consensus/ethash
panic: Log in goroutine after TestRemoteMultiNotify has completed

goroutine 460 [running]:
testing.(*common).logDepth(0xc00010a300, 0xc0001e8000, 0x69, 0x3)
        /home/ia/go1.13.5.linux-amd64/go/src/testing/testing.go:678 +0x406
testing.(*common).log(...)
        /home/ia/go1.13.5.linux-amd64/go/src/testing/testing.go:658
testing.(*common).Logf(0xc00010a300, 0x98c0d1, 0x2, 0xc000704de0, 0x1, 0x1)
        /home/ia/go1.13.5.linux-amd64/go/src/testing/testing.go:693 +0x7e
github.com/ethereum/go-ethereum/internal/testlog.(*logger).flush(0xc000146c90)
        /home/ia/go/src/github.com/ethereum/go-ethereum/internal/testlog/testlog.go:139 +0xfd
github.com/ethereum/go-ethereum/internal/testlog.(*logger).Warn(0xc000146c90, 0x99b3a4, 0x23, 0xc0006322c0, 0x4, 0x4)
        /home/ia/go/src/github.com/ethereum/go-ethereum/internal/testlog/testlog.go:104 +0xeb
github.com/ethereum/go-ethereum/consensus/ethash.(*Ethash).Seal.func2(0x0, 0xc0000365a0, 0xc000036600, 0x0, 0xc000104380, 0xc0000f6630, 0x0, 0x0, 0xc0001b0360)
        /home/ia/go/src/github.com/ethereum/go-ethereum/consensus/ethash/sealer.go:114 +0x3c5
created by github.com/ethereum/go-ethereum/consensus/ethash.(*Ethash).Seal
        /home/ia/go/src/github.com/ethereum/go-ethereum/consensus/ethash/sealer.go:103 +0x2a6
FAIL    github.com/ethereum/go-ethereum/consensus/ethash        0.169s
FAIL
```

--- a/consensus/ethash/sealer_test.go
+++ b/consensus/ethash/sealer_test.go
@@ -97,16 +97,22 @@ func TestRemoteMultiNotify(t *testing.T) {
 	ethash.config.Log = testlog.Logger(t, log.LvlWarn)
 	defer ethash.Close()
 
+	// Provide a results reader.
+	// Otherwise the unread results will be logged asynchronously
+	// and this can happen after the test is finished, causing a panic.
+	results := make(chan *types.Block, cap(sink))
+
 	// Stream a lot of work task and ensure all the notifications bubble out.
 	for i := 0; i < cap(sink); i++ {
 		header := &types.Header{Number: big.NewInt(int64(i)), Difficulty: big.NewInt(100)}
 		block := types.NewBlockWithHeader(header)
-		ethash.Seal(nil, block, nil, nil)
+		ethash.Seal(nil, block, results, nil)
 	}
 
 	for i := 0; i < cap(sink); i++ {
 		select {
 		case <-sink:
+			<-results
 		case <-time.After(10 * time.Second):
 			t.Fatalf("notification %d timed out", i)
 		}

